packages/dataviews/src/dataform.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import type { FormEventHandler } from 'react'; // TODO: how to import FormEventHandler?
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { NormalizedField } from './types';
+
+type DataFormProps< Item > = {
+	item: Item;
+	onUpdateItem: ( item: Item ) => void;
+	fields: NormalizedField< Item >[]; // TODO: use Field. Normalize them first.
+	form: {
+		closeForm: () => void;
+		isBusy: boolean;
+		onSubmitLabel?: string;
+		onCancelLabel?: string;
+	};
+};
+
+export default function DataForm< Item >( {
+	item: data,
```

Comment:
According to https://github.com/WordPress/gutenberg/issues/55101#issuecomment-2058754635 there're use cases when `DataForm` could take an array of items as input.

Reply:
Makes me wonder whether we should always pass a single item though. Even though we're batch editing a list of items, we're still showing a single value per field. I wonder if there's a use-case where we show the multiple values somehow. Anyway, not that important for now.